src/Inkluzitron/Utilities/Patiently.cs

```diff
@@ -21,15 +21,31 @@ public static async Task HandleDbConcurrency(Func<Task> concurrentDbTaskFunc)
                     await concurrentDbTaskFunc();
                     return;
                 }
-                catch (DbUpdateConcurrencyException)
+                catch (DbUpdateConcurrencyException) when (attemptNumber != Patience)
                 {
-                    if (attemptNumber == Patience)
-                        throw;
+                    await Task.Delay(BackOffDelay);
+                }
+            }
+        }
+
+        static public async Task<T> HandleDbConcurrency<T>(Func<Task<T>> concurrentDbTaskFunc)
+        {
+            if (concurrentDbTaskFunc is null)
+                throw new ArgumentNullException(nameof(concurrentDbTaskFunc));
 
-                    attemptNumber++;
+            for (var attemptNumber = 1; attemptNumber <= Patience; attemptNumber++)
+            {
+                try
+                {
+                    return await concurrentDbTaskFunc();
+                }
+                catch (DbUpdateConcurrencyException) when (attemptNumber != Patience)
+                {
                     await Task.Delay(BackOffDelay);
                 }
             }
+
+            throw new InvalidOperationException();
```

Comment:
Přidal bych sem důvod, proč je výjimka vyhozena

Reply:
Hlavně tohle je dead code, který když nastane, tak je to programátorská chyba. Ale upraveno.